### PR TITLE
2.3.x Drop dependency to javax.validation in Parameter Store and Secrets Manager 

### PR DESCRIPTION
**Pull request by MatejNedic**
Replaced javax.validation:validation-api with Validator interface implementation in spring-cloud-aws-parameter-store-config and spring-cloud-aws-secrets-manager-config.
Added both success and fail case scenarios for checking validation in both properties.

### CODE REVIEW

**Comment by MatejNedic** [comment]
Hey @maciejwalkowiak,
I can't mark pull request as ready for review. Could you review it please when you find time?

**Comment by maciejwalkowiak** [comment]
@MatejNedic sure. Thanks for contribution! I will review it in upcoming days (hopefully today but can't promise).

File: spring-cloud-aws-parameter-store-config/src/test/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertiesFailTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import org.springframework.boot.context.properties.ConfigurationPropertiesBindException;
+import org.springframework.boot.context.properties.bind.validation.BindValidationException;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.core.env.StandardEnvironment;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class AwsParamStorePropertiesFailTest {
```
[the remark this answers is not in the document]
Comment (MatejNedic) [reply spring-cloud-aws-parameter-store-config/src/test/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertiesFailTest.java:37]: You were right for @ParameterizedTest! I used it in test but wanted to ask a few questions.
Firstly is it ok to use @MethodSource and provide stream so one test simulates five test scenarios?

File: spring-cloud-aws-parameter-store-config/src/main/java/org/springframework/cloud/aws/paramstore/AwsParamStoreProperties.java [hunk 2]
```diff
@@ -66,6 +72,37 @@
...
 	private boolean enabled = true;
 
+	@Override
+	public boolean supports(Class clazz) {
+		return AwsParamStoreProperties.class.isAssignableFrom(clazz);
+	}
+
+	@Override
+	public void validate(Object target, Errors errors) {
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "prefix", "field.required",
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-parameter-store-config/src/main/java/org/springframework/cloud/aws/paramstore/AwsParamStoreProperties.java:82]: There is no need to pass array as the 4th parameter.

```
ValidationUtils.rejectIfEmptyOrWhitespace(errors, "prefix", "field.required",
				"prefix should not be empty or null.");
```

File: spring-cloud-aws-parameter-store-config/src/test/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertiesSuccessTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+package org.springframework.cloud.aws.paramstore;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Matej Nedic
+ */
+public class AwsParamStorePropertiesSuccessTest {
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-parameter-store-config/src/test/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertiesSuccessTest.java:26]: There should be single test class for success and failure. In JUnit 5 parameterized test can be just added on single method and the rest of test methods can run on regular test. So test in this class and from the one that tests failure scenarios should be moved to single `AwsParamStorePropertiesTest`.

File: spring-cloud-aws-parameter-store-config/src/test/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertiesSuccessTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+	@Test
+	void awsParamStorePropertiesAreLoaded() {
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-parameter-store-config/src/test/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertiesSuccessTest.java:29]: This test method should create valid properties object, run validator against it and verify there are no errirs.

**Pull request closed without merging**